README.md

```diff
@@ -249,6 +241,126 @@ spec:
+
+Below is an example specification for the **eks-fargate** provisioner
+
+```yaml
+apiVersion: instancemgr.keikoproj.io/v1alpha1
+kind: InstanceGroup
+metadata:
+  name: hello-world
+spec:
+  # provision for EKS using Fargate
+  provisioner: eks-fargate
+  strategy:
+    type: managed
+  # provisioner configuration
+  eks-fargate:
+    fargateProfileName: "my-unique-profile-name"
+    clusterName: "the-cluster-for-my-pods"
+    podExecutionRoleArn: "arn:aws:iam::123456789012:role/MyPodRole"
+    subnets:
+    - subnet-1a2b3c4d
+    - subnet-4d3c2b1a
+    - subnet-0w9x8y7z
+    selectors:
+    - namespace1:
+      labels:
+        key1: "value1"
+        key2: "value2"
+    - namespace2:
+      labels:
+        key1: "value1"
+        key2: "value2"
+    tags:
+      key1: "value1"
+      key2: "value2"
+
+```
+Read more about the [Fargate Profile](https://docs.aws.amazon.com/eks/latest/userguide/fargate-profile.html).
+
+If the above *podExecutionRoleArn* parameter is not specified, the provisioner will create a simple, limited role and policy that enables the pod to start but not access any AWS resources.  That role and policy is shown below. 
+
+```yaml
+Type: 'AWS::IAM::Role'
+Properties:
+  AssumeRolePolicyDocument:
+    Version: 2012-10-17
+    Statement:
+      - Effect: "Allow"
+        Principal:
+          Service: "eks-fargate-pods.amazonaws.com"
+        Action: "sts:AssumeRole"
+  ManagedPolicyArns:
+  - "arn:aws:iam::aws:policy/AmazonEKSFargatePodExecutionRolePolicy"
+  Path: /
+  
+```
+Most likely an execution role with access to addtional AWS resources will be required.  In this case, the above IAM role can be used as the basis to create a new, custom role with the IAM policies specific to your pods. Create your new role and your pod specific policies and use the new role's ARN as the *podExecutionRoleArn* parameter value in eks-fargate spec. 
+
+Here is an example of a role with an additional policy for S3 access.
+
+```yaml
+Type: 'AWS::IAM::Role'
+Properties:
+  AssumeRolePolicyDocument:
+    Version: 2012-10-17
+    Statement:
+      - Effect: "Allow"
+        Principal:
+          Service: "eks-fargate-pods.amazonaws.com"
+        Action: "sts:AssumeRole"
+  ManagedPolicyArns:
+  - "arn:aws:iam::aws:policy/AmazonEKSFargatePodExecutionRolePolicy"
+  Path: /
+  Policies:
+    - PolicyName: "your chosen name"
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action: 
+              - 's3:*'
+            Resource: '*'
+```
+
+AWS's Fargate Profiles are immutable.  Once one is created, it cannot be directly modified.  It first has to be deleted and then re-created with the desired change.
+
+The **eks-fargate** provisioner is built on top of that immutability.  Therefore, if an attempt is made to modify an existing profile, the provisioner will return an error.  You first have to `delete` the profile and follow that with a `create`.
+
+##### Bring your own role
```

Comment:
I think this section should be under `### Features` and not under `#### EKS Fargate`

Reply:
Reduced the title level from ##### to #### and it is under Features now.